cocotb/handle.py

```diff
@@ -718,7 +718,10 @@ def _check_for_set_action(self, value):
     @NonConstantObject.value.getter
     def value(self) -> BinaryValue:
         binstr = self._handle.get_signal_val_binstr()
-        result = BinaryValue(binstr, len(binstr))
+        # Skip BinaryValue.assign() as we know we are using a binstr
+        result = BinaryValue(n_bits=len(binstr))
```

Comment:
Any idea if we create BinaryValue once and return a copy (https://docs.python.org/3.5/library/copy.html) ? Not to `__init__`?

Reply:
Would that save any time? This already bypasses the slow path of `assign() -> binstr()`, which is most of the time taken up by `BinaryValue.__init__()` when you pass in a value.

It might be slightly faster to change `set_trusted_binstr()` to a classmethod and skip the deprecation checks in `__init__()`.

Reply:
You are probably be right, it is `assign`.